aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -982,6 +982,34 @@ Tensor hann_window(
       window_length, periodic, /*alpha=*/0.5, /*beta=*/0.5, options);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ kaiser_window ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tensor kaiser_window(int64_t window_length, double beta, const TensorOptions& options) {
+  return native::kaiser_window(window_length, beta, /*periodic=*/true, options);
+}
+
+Tensor kaiser_window(
+    int64_t window_length,
+    double beta,
+    bool periodic,
+    const TensorOptions& options) {
+  window_function_checks("kaiser_window", options, window_length);
+  if (window_length == 0) {
+    return at::empty({0}, options);
+  }
+  if (window_length == 1) {
+    return native::ones({1}, options);
+  }
+  if (periodic) {
+    window_length += 1;
+  }
+  auto initial = native::arange(window_length, options);
+  auto window = native::zeros(window_length, options);
```

Comment:
Why `zeros` and not `empty`? Is it because there's a dispatch issue with `native::empty`?

Reply:
Follow-up: seems like we should just use `at::empty` here